tests/unit/test_agw_network_configurator.py

```diff
@@ -0,0 +1,1037 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import os
+import time
+import unittest
+from unittest.mock import Mock, PropertyMock, call, mock_open, patch
+
+from magma_access_gateway_installer.agw_network_configurator import (
+    AGWInstallerNetworkConfigurator,
+)
+
+
+class TestAGWInstallerNetworkConfigurator(unittest.TestCase):
+    TEST_NETWORK_INTERFACES = ["eth0", "eth1"]
+
+    def setUp(self) -> None:
+        self.timestamp = time.time()
+
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator.INTERFACES_DIR"  # noqa: E501, W505
+    )
+    @patch("builtins.open", new_callable=mock_open())
+    def test_given_agw_installer_network_configurator_when_prepare_interfaces_configuration_dir_if_doesnt_exist_then_configuration_dir_gets_created_properly(  # noqa: E501
+        self, mocked_open_file, ifaces_dir
+    ):
+        ifaces_dir.return_value = f".tox/.tmp/interfaces.d_{self.timestamp}"
+        AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES, None, None
+        )._prepare_interfaces_configuration_dir_if_doesnt_exist()
+        self.assertTrue(os.path.exists(ifaces_dir))
+
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator.INTERFACES_DIR"  # noqa: E501, W505
+    )
+    def test_given_agw_installer_network_configurator_when_prepare_interfaces_configuration_dir_if_doesnt_exist_then_interfaces_configuration_dir_is_written_to_etc_network_interfaces_file(  # noqa: E501
+        self, ifaces_dir
+    ):
+        ifaces_dir.return_value = f".tox/.tmp/interfaces.d_{self.timestamp}"
+        with patch("builtins.open", mock_open()) as mocked_open_file:
+            AGWInstallerNetworkConfigurator(
+                self.TEST_NETWORK_INTERFACES, None, None
+            )._prepare_interfaces_configuration_dir_if_doesnt_exist()
+            mocked_open_file.assert_called_with("/etc/network/interfaces", "w")
```

Comment:
`/etc/network/interfaces` could be a global variable in `agw_installation_configurator.py` since it's referenced multiple time.

Reply:
I've extracted all paths which were used in both code and tests.